vpr/src/util/vpr_utils.cpp

```diff
@@ -221,17 +223,18 @@ std::string block_type_pin_index_to_name(t_physical_tile_type_ptr type, int pin_
 
     std::string pin_name = type->name;
 
-    if (type->capacity > 1) {
-        int pins_per_inst = type->num_pins / type->capacity;
-        int inst_num = pin_index / pins_per_inst;
-        pin_index %= pins_per_inst;
+    int sub_tile_index = 0;
+    auto logical_pin = type->tile_block_pin_directs_map(physical_pin(pin_index));
+    sub_tile_index = logical_pin.sub_tile_index;
+    pin_index = logical_pin.pin
```

Comment:
This code shows up in a couple places. We should probably put it into a re-usable function which does this lookup.

Reply:
I just realized that this piece of code actually won't work. The `directs_map` needs a logical block, which is not provided.

The problem here is to calculate the correct sub_tile based on a physical pin index. I think an additional member to the sub_tile defining the pin ranges so to easily calculate in which sub_tile the pin_index is located (and also change the math needed to get the pin index in the capacity instance.